modules.d/11systemd-sysext/module-setup.sh

```diff
@@ -41,13 +41,18 @@ install() {
         "$systemdsystemunitdir/systemd-confext${_suffix}.service.d/*.conf" \
         "$systemdsystemunitdir"/systemd-sysext${_suffix}.service \
         "$systemdsystemunitdir/systemd-sysext${_suffix}.service.d/*.conf" \
+        "$systemdsystemunitdir"/initrd.target.wants/systemd-confext${_suffix}.service \
+        "$systemdsystemunitdir"/initrd.target.wants/systemd-sysext${_suffix}.service \
         systemd-confext systemd-sysext
 
-    # Enable systemd type unit(s)
+    # Enable systemd type unit(s) for systemd < v258 which doesn't ship
+    # initrd.target.wants symlinks.
     for i in \
         systemd-confext.service \
         systemd-sysext.service; do
-        $SYSTEMCTL -q --root "$initdir" enable "$i"
+        if [[ -e "$initdir$systemdsystemunitdir"/"$i" ]]; then
```

Comment:
Shouldn't that check for the absence of the `initrd.target.wants` symlinks instead?

Reply:
These 2 services could run in the initrd and in the host before systemd-v258, before the initrd-specific units were added, so there were no `initrd.target.wants` symlinks before. With `_suffix="-initrd"`  we control that they are not present in `$initdir`, and if they are present (`_suffix=`), it means there are no initrd-specific units so these 2 services must be enabled.

Reply:
Thanks. The commit message mislead me. IMO those two changes should be split into two separate commits.